$OpenBSD: patch-pigz_c,v 1.2 2021/01/31 02:49:36 kn Exp $

Revert an unconditional define of _POSIX_C_SOURCE introduced in pigz 2.5
which causes <sys/cdefs.h> to define __POSIX_VISIBLE and therefore
(because _BSD_SOURCE is not defined) also define __BSD_VISIBLE 0
which prevents <unistd.h> from prototyping pledge(2).

Use pledge.

Index: pigz.c
--- pigz.c.orig
+++ pigz.c
@@ -333,7 +333,9 @@
 // Portability defines.
 #define _FILE_OFFSET_BITS 64            // Use large file functions
 #define _LARGE_FILES                    // Same thing for AIX
+#ifdef __MINGW32__
 #define _POSIX_C_SOURCE 200809L         // For MinGW
+#endif
 
 // Included headers and what is expected from each.
 #include <stdio.h>      // fflush(), fprintf(), fputs(), getchar(), putc(),
@@ -4540,6 +4542,11 @@ int main(int argc, char **argv) {
     char *opts, *p;                 // environment default options, marker
     ball_t err;                     // error information from throw()
 
+    if (pledge("stdio rpath wpath cpath fattr chown", NULL) == -1) {
+        complain("pledge");
+        exit(1);
+    }
+
     g.ret = 0;
     try {
         // initialize globals
@@ -4646,6 +4653,12 @@ int main(int argc, char **argv) {
             else if (option(argv[n]))   // process argument
                 argv[n] = NULL;         // remove if option
         option(NULL);                   // check for missing parameter
+
+        if (g.pipeout || g.decode == 2)
+            if (pledge("stdio rpath cpath", NULL) == -1) {
+                complain("pledge");
+                exit(1);
+            }
 
         // process command-line filenames
         done = 0;
